### PR TITLE
[Clusters] MVP

### PR DESCRIPTION
**Pull request by mhkshum**
# data.ts #
- created new parsed data format for clusters data, which has additional columns

# Clusters Component (Clusters.svelte) #
- created new Clusters component
- created basic html and css layout for Clusters component
- added new subheaders in all-caps

### Related Queries column ###
- wrap text after 2 rows and show ellipsis
- indent second row of text by 6 px

### Change column ###
- +/- changes
- "NEW" when change is infinity

image

### Interest column ###
- rounds to 2 decimal places

image

## Mobile View ##
- Clusters: top clusters list on top of rising clusters list when viewport < 800px
- Top Queries: top queries list is also now above rising queries list for the Top Queries component

image

### CODE REVIEW

File: src/data.ts [hunk 1]
```diff
@@ -552,4 +553,105 @@ export function fetchTopLevelQueries(selectedCountryCode): Promise<Map<string, Q
...
+}
+
+function createCluster(clusterRow: ClusterRow): Cluster {
+  const historyList = clusterRow.history === "" ? [] : clusterRow.history.split("|").map(value => Number(value));
+  const membersList = clusterRow.members === "" ? [] : removeDuplicate(clusterRow.query, clusterRow.members.split("|"));
+  const clusterChange = calculateChange(historyList, clusterRow.sni);
+  return { query: clusterRow.query, sni: clusterRow.sni, rank: clusterRow.rank, change: clusterChange, members: membersList };
+}
+
+let clustersStoragePrefix: string = "https://storage.googleapis.com/covid19-open-data/covid19-vaccination-search-insights/staging/top_clusters/";
```
Comment (mimisun) [inline src/data.ts:606]: remove "staging" from the url

Comment (mimisun) [inline src/data.ts:606]: remove "staging" from the url

**Review by mimisun — COMMENTED** [review]
Thank you Megan!

File: src/data.ts (hunk 1 above)
Comment (mhkshum) [reply src/data.ts]: Removed and can confirm that everything is showing up as expected!

image

File: src/Clusters.svelte [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+<div id="top-queries">
+    <div class="queries-header">
+        <div class="queries-category-selector">
+            <button
+                id="covid19_vaccination"
+                class={selectedListId == "covid19_vaccination"
+                    ? "map-trend-selector-button map-trend-selector-selected"
+                    : "map-trend-selector-button"}
+                on:click={changeCategory}
+                title="Search interest in any aspect of COVID-19 vaccination. For example, “when can i get the covid vaccine” or “cdc vaccine tracker”. A scaled value that you can compare across regions and times. This parent category includes searches from the other two subcategories."
```
[the remark this answers is not in the document]
Comment (mhkshum) [reply src/Clusters.svelte]: Oh yes! that is much cleaner. Modified!

File: src/Clusters.svelte [hunk 3]
```diff
@@ -0,0 +1,464 @@
...
+    let dateKey: string = "";
+    let dateRange: string = "";
+    let topLevelData: Map<string, Cluster[]> = new Map<string, Cluster[]>();
+    let countyData: Map<string, Cluster[]>;
+    let placeId: string;
+    let topQueriesList = [];
+    let risingQueriesList = [];
+    let currentSubRegion: string = "";
+
+    function capitaliseFirstLetter(str: string): string {
```
[the remark this answers is not in the document]
Comment (mhkshum) [reply src/Clusters.svelte:56]: Hi Mimi! thanks so much for reviewing everything! I couldn't figure out exactly what you were pointing to that needs capitalisation. Let me know and I can follow up and modify! thanks again!

File: src/Clusters.svelte [hunk 4]
```diff
@@ -0,0 +1,464 @@
...
+            .map((member) => capitaliseFirstLetter(member))
+            .join(", ");
+    }
+
+    function formatChange(change: number): string {
+        if (change === null) {
+            return "-";
+        }
+        if (change === Infinity) {
+            return "NEW";
```
[the remark this answers is not in the document]
Comment (mhkshum) [reply src/Clusters.svelte]: Just added! Really great catch!

**Review by mimisun — APPROVED** [review]

**Pull request merged**